test/test_sigridci_runner.py

```diff
@@ -403,6 +406,66 @@ def testDoNotValidateMetadataFileIfNotPresent(self):
 
         self.assertEqual(LOG_HISTORY, expectedLog)
         
+    def testDoNotGenerateFileWithoutEnvironmentVariables(self):
+        tempDir = tempfile.mkdtemp()
+        uploadOptions = UploadOptions(sourceDir=tempDir)
+        
+        runner = SigridCiRunner()
+        runner.prepareMetadata(uploadOptions)
+        
+        self.assertEqual(os.path.exists(f"{tempDir}/sigrid-metadata.yaml"), False)
+        
+    def testDoGenerateFileIfEnvironmentVariablesAreUsed(self):
+        tempDir = tempfile.mkdtemp()
+        uploadOptions = UploadOptions(sourceDir=tempDir)
+        os.environ["externalid"] = "1234"
+        
+        runner = SigridCiRunner()
+        runner.prepareMetadata(uploadOptions)
+        
+        self.assertEqual(os.path.exists(f"{tempDir}/sigrid-metadata.yaml"), True)
+        with open(f"{tempDir}/sigrid-metadata.yaml") as f:
+            self.assertEqual(f.read(), "metadata:\n  externalID: \"1234\"\n")
+            
+    def testIgnoreEmptyEnvironmentVariables(self):
```

Comment:
How about when in the Github action, a parameter is set to `null`? E.g., when someone wants to _delete_ the external ID, I'd expect that this `sigrid-metadata.yaml` is generated:
```yaml
metadata:
  externalID: null
```

Do we need a test for that?

Reply:
Eh… fair point, because I don't know if GitHub Actions even allows parameters to be `null`. How do we handle empty strings in the back-end? Are they considered the same as `null`, or will it actually set the external ID to an empty string?

Reply:
@patveck I have doubts if we should do this. In GitHub Actions, people do this:

```
divisionname: ${{ inputs.divisionname }}
```

Their notation is glorified Bash, so there is no real different between "value is null" and "value is empty" and "value is not set". I think it would be too error prone to let people un-set metadata in this way. I propose to leave the ability to un-set fields programmatically to the YAML file and end point, but to not support it herre.